### PR TITLE
Modify rule S6096(JavaScript): Update text to the Education Framework structure (APPSEC-231)

### PR DESCRIPTION
**Pull request by pedro-oliveira-sonarsource**

### CODE REVIEW

File: rules/S6096/javascript/how-to-fix-it/nodejs.adoc [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+[source,javascript,diff-id=1,diff-type=noncompliant]
+----
+const AdmZip = require("adm-zip");
+
+const targetDirectory = "/example/directory/";
+
+app.get('/example', (req, res) => {
+    const path = targetDirectory + req.query.filename;
+
+    const zip = new AdmZip(path); // Noncompliant
```
Comment (gaetan-ferry-sonarsource) [inline rules/S6096/javascript/how-to-fix-it/nodejs.adoc:19]: I don't really get what you are trying to do here. This looks a lot like a traditional path traversal vulnerability. Shouldn't this be a read of an uploaded zip file?
```suggestion
    const zip = new AdminZip(req.files.foo.tempFilePath); // Noncompliant
```
Also, I don't get why this specific line is marked as non compliant (except if this is for a normal path traversal).

File: rules/S6096/javascript/how-to-fix-it/nodejs.adoc [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    const zipEntries = zip.getEntries();
+
+    zipEntries.forEach(function (zipEntry) {
+        var writer = fs.createWriteStream(zipEntry.entryName);
```
Comment (gaetan-ferry-sonarsource) [inline rules/S6096/javascript/how-to-fix-it/nodejs.adoc:23]: To me, this should be the non compliant line in this case.
```suggestion
        var writer = fs.createWriteStream(zipEntry.entryName); // Noncompliant
```

File: rules/S6096/javascript/how-to-fix-it/nodejs.adoc [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+[source,javascript,diff-id=1,diff-type=compliant]
+----
+const AdmZip = require("adm-zip");
+
+const targetDirectory = "/example/directory/";
+
+app.get('/example', (req, res) => {
+    const canonicalPath = path.normalize(targetDirectory + req.query.filename);
+
+    if (canonicalPath.startsWith(targetDirectory)) {
```
Comment (gaetan-ferry-sonarsource) [inline rules/S6096/javascript/how-to-fix-it/nodejs.adoc:40]: I think this is fixing the old school path traversal issue and not the zip slip one.

File: rules/S6096/javascript/how-to-fix-it/nodejs.adoc [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        const zip = new AdmZip(canonicalPath);
+	    const zipEntries = zip.getEntries();
+
+    	zipEntries.forEach(function (zipEntry) {
+            var writer = fs.createWriteStream(zipEntry.entryName);
+            writer.write(zipEntry.getData().toString("utf8"));
+	    });
```
Comment (gaetan-ferry-sonarsource) [inline rules/S6096/javascript/how-to-fix-it/nodejs.adoc:47]: To me the zip slip issue is still there and exploitable here. Maybe refer to the existing rspec for an appropriate solution. The key is to check the entry name before extraction, not the archive name.

File: rules/S6096/javascript/how-to-fix-it/nodejs.adoc [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    }
```
Comment (gaetan-ferry-sonarsource) [inline rules/S6096/javascript/how-to-fix-it/nodejs.adoc:81]: Same remarks as above. The issue is not in the right location.

**Review by gaetan-ferry-sonarsource — CHANGES_REQUESTED** [review]
I think there is a misunderstanding of the issue to raise here. The code examples focus on a classical path traversal issue while the rule is specifically about zip slip.

**Comment by pedro-oliveira-sonarsource** [comment]
Think it's required to add the following?
const fs = require('fs');
const path = require('path');

**Review by gaetan-ferry-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**